### PR TITLE
Removing host and improving API code

### LINKED ISSUE #98: Spamming `Next` button in the game leads to code rendering lag
In the above video, I clicked `Next` about 20 times. It seems that if `Next` is clicked too frequently, the code rendered in the editor changes uncontrollably till it is consistent with the # of times `Next` is clicked. This is due to the latency of the server's response to the `on_pick` event.  Adding ratelimiting (X clicks per Y secs) should allow the host to fast forward unwanted chunks while preventing spam.

### LINKED ISSUE #137: Add support for the host to transfer their role to other players

### PR DESCRIPTION
**Pull request by Ramko9999**
Screw it, let me just commit without tests. I will manually test things out. After deploying, need to drop tables in DB and recreate the tables.
- Move DB logic from models.py to db/
- Cleaned up WS event handling
- Removed host from gitgame
- Closes #137 #98 since the concept of a host itself is removed

https://github.com/rohanshiva/gitgame/assets/41026856/5dead76e-f5bf-400a-a0be-3bc9fe6e85d9

### CODE REVIEW

File: api/db/crud.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+
+async def __has_files_to_pick(session_id: str):
+    return await File.filter(session_id=session_id).count() > 0
+
+
+async def __pick_next_file_for_code(session_id: str):
+    authors = (
+        await File.filter(session_id=session_id)
+        .annotate(count=Count("id"))
```
Comment (rohanshiva) [inline api/db/crud.py:110]: What does this do? `Count`

File: api/db/views.py
Comment (rohanshiva) [inline api/db/views.py:1]: I think I understand what `views` are, but to me these seem like they can be included in `crud.py`. What do you think? Please feel free to just ignore this, it doesn't even matter.

**Review by rohanshiva — APPROVED** [review]
lgtm, will test locally as well, but feel free to merge

thanks! 🙏🏿

File: api/db/crud.py (hunk 1 above)
Comment (Ramko9999) [reply api/db/crud.py:110]: Lines 108-112 replicate the following SQL: 

```sql
select author_id, count(distinct id) from file where session_id=SESSION_ID group by author_id
```
`__pick_next_file_for_code`'s purpose is to pick a random file created by an author. In order to make the random selection unweighted with respect to the # of files created per each author, we first randomly select the author, then randomly select a repo corresponding to the author and then randomly select the file.

The above SQL returns all the authors who have a corresponding file in the Session.

File: api/db/views.py
Comment (Ramko9999) [reply api/db/views.py:1]: Imo, the purpose of `views` is to render a resource, where as the purpose of `crud` is to enact operations on the DB. I prefer to keep them separate.

**Pull request merged**